src/jit/gentree.h

```diff
@@ -598,9 +598,9 @@ struct GenTree
 
 private:
     union {
-        // NOTE: After LSRA, one of these values may be valid even if GTF_REG_VAL is not set in gtFlags.
-        // They store the register assigned to the node. If a register is not assigned, _gtRegNum is set to REG_NA
+        // These store the register assigned to the node. If a register is not assigned, _gtRegNum is set to REG_NA
         // or _gtRegPair is set to REG_PAIR_NONE, depending on the node type.
+        // For the LEGACY_BACKEND,these are valid only if GTF_REG_VAL is set in gtFlags.
         regNumberSmall _gtRegNum;  // which register      the value is in
```

Comment:
Unrelated, but we need to do a better job of extracting regPair stuff (e.g., regPairNoSmall) from the RyuJIT build. E.g., here arm32 regNumberSmall is `char` but the unused `regPairNoSmall` is `short`, so we're (possibly) wasting space.